Hazel/ConnectionListener.cs

```diff
@@ -79,10 +79,6 @@ protected void InvokeNewConnection(MessageReader msg, Connection connection)
                 {
                 }
             }
-            else
```

Comment:
This change seems unrelated and wrong.

Reply:
While it is a bit out of scope, it's impossible to pass the tests with DEBUG mode without it. [InvokeNewConnection](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/ConnectionListener.cs#L68) [recycles](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/ConnectionListener.cs#L84) the hello reader, it's only ever called by [ReadCallback](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/Udp/UdpConnectionListener.cs#L225) which in turn always calls [HandleReceive](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/Udp/UdpConnectionListener.cs#L229) which always [disposes](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/Udp/UdpConnection.cs#L150) second time causing an [exception](https://github.com/willardf/Hazel-Networking/blob/981370c1bf67d2063b2aba9d7dbb17f66f796c7a/Hazel/ObjectPool.cs#L102-L104) every time a connection connects without anything subscribed to `NewConnection` event.

Reply:
Alright yeah, I'm on board. This is just something I missed from my breaking change. Interestingly it doesn't break any test or get captured in any way, but it is a good change.